extensions/rich_text_components/Video/Video.js

```diff
@@ -34,6 +35,33 @@ oppia.directive('oppiaNoninteractiveVideo', [
         $scope.videoUrl = $sce.trustAsResourceUrl(
           'https://www.youtube.com/embed/' + $scope.videoId + '?rel=0' +
           $scope.timingParams + $scope.autoplaySuffix);
+
+        $scope.videoWidth = 560;
+        $scope.videoHeight = 315;
+
+        var aspectRatio = $scope.videoHeight / $scope.videoWidth;
+
+        var tutorCard = $(".conversation-skin-main-tutor-card");
+        tutorCard.addClass('contains-video');
+
+        var outerPadding = 60;
+
+        var resizeVideo = function() {
+          var newWidth = tutorCard.width() - outerPadding;
+
+          $scope.videoWidth = newWidth;
+          $scope.videoHeight = newWidth * aspectRatio;
+        }
+
+        angular.element($window).bind('resize', resizeVideo);
+        $timeout(function() {
+          resizeVideo();
+        }, 200);
+
+        $element.on('$destroy', function() {
```

Comment:
Yeah, this is a bit ick. This sort of logic should go in the skin controller; maybe you could have a function along the lines of doesContentHtmlContainVideo()